### PR TITLE
Secure static content

### LINKED ISSUE #1654: Documentation request. How to secure content with StaticContentSupport
This should be a really easy one, but I can't seem to find a solution. How do you secure content registered with StaticContentSupport? I'd like to have /protected static content and open static content, with Routing looking something like this:

return Routing.builder()
.register(WebSecurity.create(getSecurity()))
.register("/protected", StaticContentSupport.builder(protectedPath).welcomeFileName("index.html").build())
.register("/", StaticContentSupport.builder(openPath).welcomeFileName("index.html").build())
.error(Exception.class, (req, res, ex) -> {
    res.status(Http.Status.NOT_FOUND_404).send();
})
.build();

What else is required to secure /protected?

### PR DESCRIPTION
**Pull request by tomas-langer**
Example and documentation for securing non-JAX-RS endpoint using Helidon security.

Resolves #1654

### CODE REVIEW

**Review by Verdent — APPROVED** [review]
LGTM

File: docs/mp/security/03_configuration-secrets.adoc [hunk 1]
```diff
@@ -27,35 +27,7 @@ When security requires a configuration with repeating complex elements, use Heli
...
-  web-server:
-    paths:
-      - path: "/static-cp[/{*}]"
-        authenticate: true
-----
-
-=== Protecting Configuration Secrets 
+== Protecting Configuration Secrets
 
 In Helidon MP, the config encryption filter ** enabled by default**. However, if you don't configure it, the filter
```
Comment (ljamen) [inline docs/mp/security/03_configuration-secrets.adoc:32]: the config encryption filter is

File: docs/shared/security/helidon-endpoints.adoc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+The following section describes configuration of such protection using configuration files,
+ in this case using a `yaml` file, as it provides a tree structure.
+
+==== Configuring endpoint protection
+
+The configuration is usually placed under `security.web-server` (this can be
+customized in Helidon SE).
+
+The following example shows an example we will explain in detail:
```
Comment (ljamen) [inline docs/shared/security/helidon-endpoints.adoc:41]: The following shows an example we will explain in detail:

File: docs/shared/security/helidon-endpoints.adoc [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+<1> Attribute based access control provider that checks roles and scopes
+<2> The provider(s) used in your application, such as `oidc`
+<3> Default configuration for all configured paths
+<4> Protection of `/metrics` and all nested paths with `admin` role required
+<5> Protection of `/health` and all nested paths with `monitor` role required
+<6> Protection of `/openapi` and all nested paths with `openapi` scope required
+<7> Protection of static content configured on `/static` path with either `user` or `monitor` role required
+
+If you need to use a properties file, such as for `microprofile-config.properties`, you
```
Comment (ljamen) [inline docs/shared/security/helidon-endpoints.adoc:73]: such as `microprofile-config.properties`,

File: examples/security/basic-auth-with-static-content/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Web Server Integration and Basic Authentication
+
+This example demonstrates Integration of WebServer
```
Comment (ljamen) [inline examples/security/basic-auth-with-static-content/README.md:3]: Web Server

**Review by ljamen — APPROVED** [review]
Only minor fixes.

**Pull request merged**